File: Games/EchoGame/src/main/java/jokatu/game/games/echo/game/EchoGame.java
package jokatu.game.games.echo.game;

import jokatu.game.Game;
import jokatu.game.GameID;
import jokatu.game.event.StatusUpdateEvent;
import jokatu.game.player.StandardPlayer;
import org.jetbrains.annotations.NotNull;

import java.util.Timer;
import java.util.TimerTask;

import static java.time.LocalTime.now;
import static java.time.format.DateTimeFormatter.ofPattern;

public class EchoGame extends Game<StandardPlayer> {

	public static final String ECHO = "Echo";

	EchoGame(GameID identifier) {
		super(identifier);
		new Timer().scheduleAtFixedRate(
				new TimerTask() {
					@Override
					public void run() {
						fireEvent((StatusUpdateEvent) () -> () -> "The time is: " + now().format(ofPattern("HH:mm:ss")));
					}
				},
				0,
				1000
		);
	}

	@NotNull
	@Override
	public String getGameName() {
		return ECHO;
	}

	@Override
	public void advanceStageInner() {
		currentStage = new EchoStage();
	}
}

File: Games/EchoGame/src/main/kotlin/jokatu/game/games/echo/game/EchoGame.kt
package jokatu.game.games.echo.game

import jokatu.game.Game
import jokatu.game.GameID
import jokatu.game.player.StandardPlayer
import jokatu.game.stage.machine.SingleStageMachine
import jokatu.game.status.StandardTextStatus
import java.time.LocalTime.now
import java.time.format.DateTimeFormatter.ofPattern
import java.util.*

class EchoGame internal constructor(identifier: GameID) : Game<StandardPlayer>(identifier) {

	override val gameName: String
		get() = ECHO

	init {
		stageMachine = SingleStageMachine(EchoStage())

		val status = StandardTextStatus()
		status.observe(::fireEvent)

		Timer().scheduleAtFixedRate(
				object : TimerTask() {
					override fun run() {
						status.text = "The time is: ${now().format(ofPattern("HH:mm:ss"))}"
					}
				},
				0,
				1000
		)
	}

	companion object {
		const val ECHO = "Echo"
	}
}
